text/0005-application-formalization.md

```diff
@@ -0,0 +1,194 @@
+| version  | String  | Version of the module                                                | e.g. "13.5.0"                    |
+
+**ModuleId Properties**
+| Property | Type    | Description                                                          | Notes                            |
+| -------- | ------- | -------------------------------------------------------------------- | -------------------------------- |
+| id       | String  | Identifier for the module conforming to pattern: `{name}-{version}`  | e.g. "mod-orders-storage-13.5.0" |
+| name     | String  | Name of the module                                                   | e.g. "mod-orders"                |
+| version  | String  | Version of the module                                                | e.g. "13.5.0"                    |
+| url      | String  | URL pointing to the module descriptor for this module                | e.g. "https://folio-registry.dev.folio.org/_/proxy/modules/mod-orders-storage-13.5.0" |
+
+Example:
+```
+{
+  "id": "app-gobi-0.0.1",
+  "name": "app-gobi",
+  "version": "0.0.1",
+  "description": "Application facilitating the placement of orders via GOBI and other compatible vendors",
+  "dependencies": [
+    {
+      "id": "app-orders-0.0.1",
+      "name": "app-orders",
+      "version": "^0.0.1"
+    },
+    {
+      "id": "app-organizations-0.0.1",
+      "name": "app-organizations",
+      "version": "^0.0.1"
+    }
+  ],
+  "modules": [
+    {
+      "id": "mod-gobi-2.6.0",
+      "name": "mod-gobi",
+      "version": "2.6.0",
+	  "url": "https://folio-registry.dev.folio.org/_/proxy/modules/mod-gobi-2.6.0"
+    }
+  ],
+  "uiModules": [
+    {
+      "id": "folio_gobi-settings-2.0.0",
+      "name": "folio_gobi-settings",
+      "version": "2.0.0",
+	  "url": "https://folio-registry.dev.folio.org/_/proxy/modules/folio_gobi-settings-2.0.0"
+    }
+  ]
+}
+```
+**N.B.** Full module descriptors and metadata have been omitted for readability reasons.
+
+## Benefits
+* This is the first step toward breaking the monolithic release into application-specific releases
+* Results in fewer things to work with at the system operator level
+* Brings us closer to realizing the idea of having an application store/marketplace
+* Allows for introducing/expanding the bounded context (subject of a separate RFC)
+
+## Risks and Drawbacks
+* Folio has too many module dependencies which must be reduced
+* Transition to Applications will need to be iterative
+  * First pass is to create larger applications spanning multiple areas in order to satisfy dependencies
+  * Subsequent passes will further break up these large applications into smaller, applications focused on a single area of functionality
+* Since Applications are a vertical slice of functionality, better alignment/coordination between back-end and front-end development teams is required.
+  * Reorganizing development teams around applications would help
```

Comment:
Several FOLIO governmental decisions - reorganizing development teams is just one of them - need to be made and come into effect, in order to make the application formalizaion possible. Those are not part of this RFC ?

Reply:
I believe these topics will be addressed by the Tri-council application formalization working group.

Reply:
Agree with Craig that we've delegated the governance pieces to the governance group.

Reply:
@inkuss Please LMK if it's OK to mark this thread resolved, or if further discussion is required